octobot_backtesting/time/time_manager.py

```diff
@@ -47,14 +51,32 @@ def start(self):
 
         self.current_timestamp = self.starting_timestamp
 
-    def __reset_time(self):
+    def _reset_time(self):
         self.set_current_timestamp(0)
 
     def has_finished(self):
+        if self.timestamps_whitelist is not None:
+            if self._timestamps_whitelist_index >= len(self.timestamps_whitelist) - 1:
+                return True
         return self.current_timestamp >= self.finishing_timestamp
 
     def next_timestamp(self):
         self.current_timestamp += self.time_interval
+        if self.timestamps_whitelist is not None:
+            # when timestamps_whitelist is set: fast forward time to only trigger whitelisted timestamps
+            while self._should_skip_current_timestamp() and self.current_timestamp <= self.finishing_timestamp:
+                self.current_timestamp += self.time_interval
+
+    def _should_skip_current_timestamp(self):
+        if self.timestamp_accept_check_callback is not None and self.timestamp_accept_check_callback():
+            return False
+        return not self._has_current_timestamp_in_whitelist()
+
+    def _has_current_timestamp_in_whitelist(self):
+        while self.timestamps_whitelist[self._timestamps_whitelist_index] < self.current_timestamp and \
+             self._timestamps_whitelist_index < len(self.timestamps_whitelist) - 1:
+            self._timestamps_whitelist_index += 1
+        return self.timestamps_whitelist[self._timestamps_whitelist_index] == self.current_timestamp
 
```

Comment:
I think we could use a queue instead of `timestamps_whitelist` and `_timestamps_whitelist_index`. We could dequeue here and checking if the queue is empty in `has_finished`

Reply:
We need to access this list in scripts therefore I think it's good to keep it somewhere. I'm add the queue queue to make the usage of this list simpler and to remove _timestamps_whitelist_index